From b4aeb6ee25a484f2f3a9e64e98c9cc0d95e52ae1 Mon Sep 17 00:00:00 2001
From: Sean Cross <xobs@kosagi.com>
Date: Fri, 19 Feb 2016 12:10:41 +0800
Subject: [PATCH 56/65] Revert "serial: imx: don't enable clocks on
 suspend/resume"

This reverts commit a0e052879bac09d87940662934e7a868739f297c.  It is
no longer required.
---
 drivers/tty/serial/imx.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index dc2cf2a..016e4be 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -2069,9 +2069,16 @@ static int imx_serial_port_suspend_noirq(struct device *dev)
 {
 	struct platform_device *pdev = to_platform_device(dev);
 	struct imx_port *sport = platform_get_drvdata(pdev);
+	int ret;
+
+	ret = clk_enable(sport->clk_ipg);
+	if (ret)
+		return ret;
 
 	serial_imx_save_context(sport);
 
+	clk_disable(sport->clk_ipg);
+
 	return 0;
 }
 
@@ -2079,9 +2086,16 @@ static int imx_serial_port_resume_noirq(struct device *dev)
 {
 	struct platform_device *pdev = to_platform_device(dev);
 	struct imx_port *sport = platform_get_drvdata(pdev);
+	int ret;
+
+	ret = clk_enable(sport->clk_ipg);
+	if (ret)
+		return ret;
 
 	serial_imx_restore_context(sport);
 
+	clk_disable(sport->clk_ipg);
+
 	return 0;
 }
 
-- 
2.7.3

